cpp/arcticdb/stream/schema.hpp

```diff
@@ -25,7 +25,11 @@ class FixedSchema {
   public:
     FixedSchema(StreamDescriptor desc, const Index& index) :
         desc_(std::move(desc)),
-        index_(std::move(index)){}
+        index_(index){}
+
+    static FixedSchema default_schema() {
+        return FixedSchema(StreamDescriptor(), stream::TimeseriesIndex::default_index());
+    }
```

Comment:
They are served merely with `variant_match` to runtime dispatch template parameters. The choice of index carries no meaning. It references `inline Index default_index_type_from_descriptor(const IndexDescriptor::Proto &desc);`

Reply:
I still think it's a bit of a code smell. If the index type matters then we are setting it wrongly, as it could be a row count index or a string index. If it doesn't matter, why are we setting it at all?

Reply:
Could we pass the index type in as an argument?